src/ChainSafe.GamingSDK.EVM/Web3/Core/Web3.cs

```diff
@@ -82,7 +83,10 @@ public void Terminate()
 
             if (initialized)
             {
-                // todo terminate other components
+                foreach (var lifecycleParticipant in serviceProvider.GetServices<ILifecycleParticipant>())
```

Comment:
same with this

Reply:
Gonna resolve this one, let's continue the discussion on the other one